### ISSUE TITLE
add `.default()` method

### ISSUE THREAD

**Issue by jonschlinkert** [jonschlinkert/cache-base#9]
I think it would be nice to support adding default values that would work like this:

```js
const CacheBase = require('cache-base');
const app = new CacheBase();

app.set('foo', 'xxx');
app.default('foo', 'one');
app.default('bar', 'two');
app.default('baz', 'three');
app.set('baz', 'zzz');

console.log(app.get('foo'));
//=> 'xxx'

console.log(app.get('bar'));
//=> 'two'

console.log(app.get('baz'));
//=> 'zzz'

console.log(app);
// Cache {
//   cache: { foo: 'xxx', bar: 'two', baz: 'zzz' },
//   defaults: { foo: 'one', bar: 'two', baz: 'three' } }
```

@doowb any thoughts?

Another issue, https://github.com/jonschlinkert/cache-base/issues/5 describes a similar but different approach

**Comment by doowb** [comment]
I like this way of doing defaults. I think we already implemented something like this in `option-cache`.

**Comment by jonschlinkert** [comment]
closed by 3.0
